src/lib/containers/entity/page/action_menu/EntityActionMenu.tsx

```diff
@@ -0,0 +1,234 @@
+import React from 'react'
+import { SxProps } from '@mui/material'
+import { ComplexMenu, IconButtonConfiguration } from '../../../menu/ComplexMenu'
+import IconSvg, { Icon } from '../../../IconSvg'
+import { DropdownMenuItem, DropdownMenuProps } from '../../../menu/DropdownMenu'
+
+// Represents the two types of dropdown menus that will be displayed on the entity page
+type EntityActionMenuDropdownMenuType = 'DOWNLOAD' | 'PRIMARY'
+
+type EntityActionMenuDropdownMenuConfiguration = {
+  visible: boolean
+  tooltipText?: string
+  disabled?: boolean
+}
+
+export type ActionConfiguration = {
+  visible: boolean
+  text: string
+  disabled?: boolean
+  tooltipText?: string
+  onClick?: (e: React.MouseEvent) => void
+  href?: string
+}
+
+export type ActionConfigurationMap = Record<string, ActionConfiguration>
+
+export type MenuConfigurationMap = Record<
+  EntityActionMenuDropdownMenuType,
+  EntityActionMenuDropdownMenuConfiguration
+>
+
+export type ActionViewProps = {
+  action: string
+  icon?: Icon
+  textSx?: SxProps
+  iconSx?: SxProps
+}
+
+export type EntityActionMenuLayout = {
+  buttonActions: ActionViewProps[]
+  downloadMenuActions: ActionViewProps[][]
+  primaryMenuActions: ActionViewProps[][]
+  primaryMenuText: string
+  primaryMenuEndIcon: Icon
+  menuButtonSx?: SxProps
```

Comment:
While text/icon/sx probably make more sense to include in the `MenuConfigurationMap` object, these only change when the layout changes, so it was easier to just include them here.